### PR TITLE
[openshift-groups] introduce new integration

### PR DESCRIPTION
**Pull request by maorfr**
this integration does:
* add users to groups
* delete users from groups

this integration does not:
* manage all groups (only ones specified in a cluster file under `managedGroups`)
* creates groups
* deletes groups

### CODE REVIEW

File: reconcile/openshift_groups.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+
+
+def fetch_desired_state():
+    gqlapi = gql.get_api()
+    roles = gqlapi.query(ROLES_QUERY)['roles']
+    desired_state = []
+
+    for r in roles:
+        for p in r['permissions']:
+            if 'service' not in p:
```
Comment (jmelis) [inline reconcile/openshift_groups.py]: a bit redundant?

Comment (jmelis) [reply reconcile/openshift_groups.py]: Since we already have the one below...

File: reconcile/openshift_groups.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+def subtract_states(from_state, subtract_state, action):
+    result = []
+
+    for f_user in from_state:
+        found = False
+        for s_user in subtract_state:
+            if f_user != s_user:
+                continue
+            found = True
+            break
```
Comment (jmelis) [inline reconcile/openshift_groups.py]: maybe this is clearer?

```
        for s_user in subtract_state:
            if f_user == s_user:
                found = True
                break
```

**Comment by jmelis** [comment]
Excellent stuff @maorfr !!

File: reconcile/openshift_groups.py (hunk 1 above)
Comment (maorfr) [reply reconcile/openshift_groups.py]: right!

File: reconcile/openshift_groups.py (hunk 2 above)
Comment (maorfr) [reply reconcile/openshift_groups.py]: Flat is better than nested? ;)

**Comment by maorfr** [comment]
for future reference -
https://wiki.jenkins.io/display/JENKINS/Role+Strategy+Plugin

**Pull request merged**